index.js

```diff
@@ -84,16 +84,36 @@ function typeis (value, types_) {
  * or `content-length` headers set.
  * http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.3
  *
+ * A http/2 request with DataFrame can have no `content-length` header.
+ * https://httpwg.org/specs/rfc7540.html
+ *
+ * A http/2 request without DataFrame send HeaderFrame with end-stream-flag.
+ * If nodejs gets end-stream-flag, then nodejs ends readable stream.
+ * https://github.com/nodejs/node/blob/master/lib/internal/http2/core.js#L301
+ *
  * @param {Object} request
  * @return {Boolean}
  * @public
  */
 
 function hasbody (req) {
-  return req.headers['transfer-encoding'] !== undefined ||
+  return (ishttp2(req) && !req.stream._readableState.ended) ||
```

Comment:
Some testing seems like this may have an issue. If you add a `req.on('data'` listener and read the body, this then returns `false` for http/2 but `true` for http/1. The current interface it is expected to be true for both, as hasBody determines if the request has body, regardless of it you happened to have read it already or not.

Reply:
I checked and happens an issue you said. I address this issue and add a test.